src/border_agent/border_agent.cpp

```diff
@@ -484,12 +516,19 @@ bool BorderAgent::IsThreadStarted(void) const
     return role == OT_DEVICE_ROLE_CHILD || role == OT_DEVICE_ROLE_ROUTER || role == OT_DEVICE_ROLE_LEADER;
 }
 
-std::string BorderAgent::BaseServiceInstanceName() const
+std::string BorderAgent::MakeServiceInstanceName() const
 {
     const otExtAddress *extAddress = otLinkGetExtendedAddress(mNcp.GetInstance());
     std::stringstream   ss;
+    std::string         instanceName = mVendorName + " " + mProductName;
+
+    // The service instance name can always be overriden by `OTBR_MESHCOP_SERVICE_INSTANCE_NAME`
+    if (strcmp(OTBR_MESHCOP_SERVICE_INSTANCE_NAME, DEFAULT_MESHCOP_SERVICE_INSTANCE_NAME) != 0)
```

Comment:
Do we really need `DEFAULT_MESHCOP_SERVICE_INSTANCE_NAME`? What about we just set the `instanceName` as `OTBR_MESHCOP_SERVICE_INSTANCE_NAME`?